src/api/token.rs

```diff
@@ -0,0 +1,17 @@
+use crate::client::LichessApi;
+use crate::error::Result;
+use crate::model::token::*;
+
+impl LichessApi<reqwest::Client> {
+    pub async fn test_tokens(
+        &self,
+        request: impl Into<test::PostRequest>,
+    ) -> Result<TokenTestResults> {
+        self.get_single_model(request.into()).await
+    }
+
+    pub async fn revoke_token(&self) -> Result<bool> {
+        // XXX get_ok expects a bool, but this method just returns 200
+        self.get_ok(revoke::DeleteRequest::new()).await
+    }
```

Comment:
I should have documented this, but `get_ok` is because for many endpoints the lichess server actually returns `{ "ok": true }`, which this helper function converts to bool. This seems to be the first method that returns nothing. It will need a new function in `api/mod.rs`. Currently everything is model based, I'm not sure if we can reuse that to make 'nothing' a model or if we need to refactor the code to allow for a 'model-less' request. If this is too much work, you could separate this out and just submit the delete endpoint for PR.

Reply:
Also, does it return 200, or 204 as per the docs?

Reply:
Tried

```diff
commit 80248de72c361209acc64642ed1f39a129683f11
Author: Ruben De Smet <ruben.de.smet@rubdos.be>
Date:   Sun Oct 13 14:29:00 2024 +0200

    Add Nothing model

diff --git a/src/api/mod.rs b/src/api/mod.rs
index 6775a6a..da0fba2 100644
--- a/src/api/mod.rs
+++ b/src/api/mod.rs
@@ -33,6 +33,17 @@ impl LichessApi<reqwest::Client> {
         return Ok(result?.ok);
     }
 
+    pub async fn get_empty<Q, B>(&self, request: Request<Q, B>) -> Result<()>
+    where
+        Q: QueryBounds,
+        B: BodyBounds,
+    {
+        let _result = self
+            .get_single_model::<Q, B, crate::model::Nothing>(request)
+            .await?;
+        return Ok(());
+    }
+
     pub async fn get_pgn<Q, B>(
         &self,
         request: Request<Q, B>,
diff --git a/src/api/oauth.rs b/src/api/oauth.rs
index 2d88739..57de6cd 100644
--- a/src/api/oauth.rs
+++ b/src/api/oauth.rs
@@ -7,8 +7,7 @@ impl LichessApi<reqwest::Client> {
         self.get_single_model(request.into()).await
     }
 
-    pub async fn revoke_token(&self) -> Result<bool> {
-        // XXX get_ok expects a bool, but this method just returns 200
-        self.get_ok(revoke::DeleteRequest::new()).await
+    pub async fn revoke_token(&self) -> Result<()> {
+        self.get_empty(revoke::DeleteRequest::new()).await
     }
 }
diff --git a/src/model/mod.rs b/src/model/mod.rs
index fc69424..91f7d8a 100644
--- a/src/model/mod.rs
+++ b/src/model/mod.rs
@@ -227,6 +227,9 @@ fn to_form_string<B: BodyBounds>(body: &B) -> error::Result<String> {
     serde_urlencoded::to_string(&body).map_err(|e| error::Error::UrlEncoded(e))
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Nothing;
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Ok {
     pub ok: bool,
```

But that was apparently too easy: Failed to revoke token: response error: empty response stream

Indeed 204.

Reply:
9112af34a16f51e243a6f7c2a5c57ba303e2d74c seems happy! :-)